pebblo/app/service/discovery/discovery_service.py

```diff
@@ -0,0 +1,259 @@
+# Discovery API with database implementation.
+from datetime import datetime
+
+from pebblo.app.enums.enums import ApplicationTypes
+from pebblo.app.libs.responses import PebbloJsonResponse
+from pebblo.app.models.db_models import (
+    AiApp,
+    AiDataLoader,
+    Chain,
+    FrameworkInfo,
+    InstanceDetails,
+    Metadata,
+    PackageInfo,
+    VectorDB,
+)
+from pebblo.app.models.models import DiscoverAIAppsResponseModel
+from pebblo.app.models.sqltables import AiAppTable, AiDataLoaderTable
+from pebblo.app.service.discovery.common import get_or_create_app
+from pebblo.app.storage.sqlite_db import SQLiteClient
+from pebblo.app.utils.utils import get_pebblo_server_version, timeit
+from pebblo.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class AppDiscover:
+    def __init__(self):
+        self.db = None
+        self.data = None
+        self.app_name = None
+
+    @staticmethod
+    def _get_current_datetime():
+        """
+        Return current datetime
+        """
+        return datetime.now().isoformat()
+
+    @staticmethod
+    def return_response(message, status_code, pebblo_server_version=None):
+        response = DiscoverAIAppsResponseModel(
+            pebblo_server_version=pebblo_server_version,
+            message=str(message),
+        )
+        return PebbloJsonResponse.build(
+            body=response.dict(exclude_none=True), status_code=status_code
+        )
+
+    def _fetch_runtime_instance_details(self) -> InstanceDetails:
+        """
+        Retrieve instance details from input data and return its corresponding model object.
+        """
+        logger.debug("Retrieving instance details from input data.")
+        # Fetching runtime instance details
+        runtime_dict = self.data.get("runtime", {})
+        instance_details_model = InstanceDetails(
+            language=runtime_dict.get("language"),
+            languageVersion=runtime_dict.get("language_version"),
+            host=runtime_dict.get("host"),
+            ip=runtime_dict.get("ip"),
+            path=runtime_dict.get("path"),
+            runtime=runtime_dict.get("runtime"),
+            type=runtime_dict.get("type"),
+            platform=runtime_dict.get("platform"),
+            os=runtime_dict.get("os"),
+            osVersion=runtime_dict.get("os_version"),
+            createdAt=self._get_current_datetime(),
+        )
+        logger.debug(f"AiApp Name [{self.app_name}]")
+        return instance_details_model
+
+    def create_app_obj(
+        self, ai_app, instance_details, chain_details, retrievals_details, app_type
+    ):
+        """
+        Create an AI App Model and return the corresponding model object
+        """
+        logger.debug("Creating App model object")
+        # Initialize Variables
+        current_time = self._get_current_datetime()
+        last_used = current_time
```

Comment:
Do we require this last_used variable

Reply:
yes, we are using it.

Reply:
We are using it but can't we use current_time variable instead.. Is last_used or current_time variable being changed somewhere?